### PR TITLE
[Enchancement](table-function) optimization for  vectorized table function

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes
before
```sql
mysql [test]>select sum(k1) from d_table  lateral  view  explode_numbers(20)  tmp1  as  e1;
+-----------+
| sum(`k1`) |
+-----------+
|         0 |
+-----------+
1 row in set (1.12 sec)
```

after
```sql
mysql [test]>select sum(k1) from d_table  lateral  view  explode_numbers(20)  tmp1  as  e1;
+-----------+
| sum(`k1`) |
+-----------+
|         0 |
+-----------+
1 row in set (0.44 sec)
```

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exprs/table_function/vexplode.cpp [hunk 1]
```diff
@@ -75,36 +75,18 @@ Status VExplodeTableFunction::reset() {
     return Status::OK();
 }
 
-Status VExplodeTableFunction::get_value(void** output) {
+void VExplodeTableFunction::get_value(MutableColumnPtr& column) {
     if (_is_current_empty) {
-        *output = nullptr;
-        return Status::OK();
+        column->insert_default();
```
Comment (HappenLee) [inline be/src/vec/exprs/table_function/vexplode.cpp:80]: should return, not do the after logic

File: be/src/vec/exprs/table_function/vexplode_numbers.cpp [hunk 2]
```diff
@@ -35,7 +37,15 @@ Status VExplodeNumbersTableFunction::process_init(vectorized::Block* block) {
     RETURN_IF_ERROR(_vexpr_context->root()->children()[0]->execute(_vexpr_context, block,
                                                                    &value_column_idx));
     _value_column = block->get_by_position(value_column_idx).column;
-
+    if (is_column_const(*_value_column)) {
+        int value = *reinterpret_cast<const int*>(_value_column->get_data_at(0).data);
```
Comment (HappenLee) [inline be/src/vec/exprs/table_function/vexplode_numbers.cpp:41]: better call get_int

File: be/src/vec/exprs/table_function/vexplode_numbers.cpp [hunk 3]
```diff
@@ -35,7 +37,15 @@ Status VExplodeNumbersTableFunction::process_init(vectorized::Block* block) {
...
+        if (value <= block->rows()) { // use const optimize
```
Comment (HappenLee) [inline be/src/vec/exprs/table_function/vexplode_numbers.cpp:42]: why the value need to <= block->rows()?

File: be/src/vec/exprs/table_function/vexplode_numbers.cpp [hunk 4]
```diff
@@ -68,22 +78,24 @@ Status VExplodeNumbersTableFunction::reset() {
...
 }
 
-Status VExplodeNumbersTableFunction::get_value(void** output) {
+void VExplodeNumbersTableFunction::get_value(MutableColumnPtr& column) {
     if (_is_current_empty) {
-        *output = nullptr;
+        if (_is_nullable) {
+            static_cast<ColumnInt32*>(
+                    static_cast<ColumnNullable*>(column.get())->get_nested_column_ptr().get())
+                    ->insert_default();
```
Comment (HappenLee) [inline be/src/vec/exprs/table_function/vexplode_numbers.cpp:86]: the work may cause the nested column size do not == null map size?

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.77 seconds
 stream load tsv:          455 seconds loaded 74807831229 Bytes, about 156 MB/s
 stream load json:         23 seconds loaded 2358488459 Bytes, about 97 MB/s
 stream load orc:          73 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230321210051_clickbench_pr_118158.html

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run arm

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/vec/columns/column_nullable.h [hunk 5]
```diff
@@ -350,6 +350,13 @@ class ColumnNullable final : public COWHelper<IColumn, ColumnNullable> {
 
     ColumnPtr index(const IColumn& indexes, size_t limit) const override;
 
+    Int64 get_int(size_t n) const override {
```
Comment (HappenLee) [inline be/src/vec/columns/column_nullable.h:353]: do not use the function，0 means two way

File: be/src/vec/exprs/table_function/vexplode_numbers.cpp [hunk 6]
```diff
@@ -35,7 +37,15 @@ Status VExplodeNumbersTableFunction::process_init(vectorized::Block* block) {
     RETURN_IF_ERROR(_vexpr_context->root()->children()[0]->execute(_vexpr_context, block,
                                                                    &value_column_idx));
     _value_column = block->get_by_position(value_column_idx).column;
-
+    if (is_column_const(*_value_column)) {
+        int value = *reinterpret_cast<const int*>(_value_column->get_data_at(0).data);
+        if (value <= block->rows()) { // use const optimize
+            _is_const = true;
+            for (int i = 1; i <= value; i++) {
+                ((ColumnInt32*)_elements_column.get())->insert_value(i);
```
[the remark this answers is not in the document]
Comment (HappenLee) [reply be/src/vec/exprs/table_function/vexplode_numbers.cpp:45]: get_data_at()

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run arm

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**